rocketpy/mathutils/function.py

```diff
@@ -2825,7 +2825,12 @@ def _check_user_input(
 
         # check source for data type
         # if list or ndarray, check for dimensions, interpolation and extrapolation
-        if isinstance(source, (list, np.ndarray)):
+        if isinstance(source, (list, np.ndarray, str, Path)):
+            # Deal with csv or txt
+            if isinstance(source, (str, Path)):
```

Comment:
I believe a `try-except` with `FileNotFound` would be beneficial here, just because it is a good practice. But again, this is also done in `set_source`.